internal/backends/selector/lib.rs

```diff
@@ -48,6 +50,61 @@ cfg_if::cfg_if! {
     }
 }
 
+pub struct PlatformBuilder {
+    opengl_api: Option<OpenGLApi>,
+    renderer: Option<SlintRenderer>,
+}
+
+impl PlatformBuilder {
+    /// Creates a new PlatformBuilder for configuring aspects of the Platform.
+    pub fn new() -> PlatformBuilder {
+        PlatformBuilder { opengl_api: None, renderer: None }
+    }
+
+    /// Configures this builder to use the specified OpenGL API when building the platform later.
+    pub fn with_opengl_api(mut self, opengl_api: OpenGLApi) -> Self {
```

Comment:
In the future I'd like to add a `GraphicsAPI` kind of enum that covers `Vulkan`, `OpenGL`, `Metal`, etc. and we could have a `with_graphics_api(...)` function in this builder as well. Using `with_opengl_api` would then imply a call to `with_graphics_api(OpenGL)`. This is just a thought for the future, regardless of this PR :)

Reply:
Yes, "how far do I go" was something I asked myself multiple times while looking into this :slightly_smiling_face:. I figured at least before getting feedback I would keep it simple, but I agree that being able to request specific graphics APIs/features would be ideal.

Something like:
```
struct ApiVersion {
  major: u8,
  minor: u8
}

enum OpenGLApi {
  GL(Option<ApiVersion>),
  GLES(Option<ApiVersion>)
}

enum GraphicsApi {
  OpenGL(OpenGLApi),
  Vulkan(VulkanApi)
}
```

or maybe it makes sense to have something like
```
struct OpenGLOptions {
  version: Option<ApiVersion>
}

struct OpenGLESOptions {
  version: Option<ApiVersion>
}

enum OpenGLApi {
  GL(Option<OpenGLOptions>),
  GLES(Option<OpenGLESOptions>)
}
```
for selecting version + more to allow more flexibility down the road :thinking: 

(It looks like there's already a [GraphicsAPI enum](https://github.com/slint-ui/slint/blob/f8f1d468a0aba1b11e7f9fb71739e46a178d4415/internal/core/api.rs#L255) for WebGL/OpenGL selection, maybe a slightly different name would be better to avoid confusion.)

I don't mind going with this approach for this PR, or we can leave it as-is for now to keep it simpler. Let me know your preference.

Reply:
I agree, I think it’s okay to leave it as-is for now. I think it fits well with a future extension.